t/t-attributes.sh

```diff
@@ -36,3 +36,22 @@ begin_test "macros"
   ! grep '"dir/*.bin" already supported' track.log
 )
 end_test
+
+begin_test "macros with HOME"
+(
+  set -e
+
+  reponame="$(basename "$0" ".sh")-home"
+  clone_repo "$reponame" repo-home
+
+  mkdir -p "$HOME/.config/git"
+  printf '[attr]lfs filter=lfs diff=lfs merge=lfs -text\n*.dat lfs\n' \
+    > "$HOME/.config/git/attributes"
+
+  contents="some data"
+  printf "$contents" > foo.dat
+  git add *.dat
+  git commit -m 'foo.dat'
+  assert_local_object "$(calc_oid "$contents")" 9
```

Comment:
You're right.  Git will already have run it through the filter, so the only case in which we wouldn't have the object would be if Git hadn't honored the setting, and we're not testing Git.

I'll leave this as it is for the purposes of ensuring general sanity, but add a call to `git lfs track`.

Reply:
Ah, one thing to mention is that on this line (unlike below), we're working solely with files outside of the working tree, which `git lfs track` intentionally ignores (except to read macros).  Below, we don't have that problem, since the file in the working tree isn't ignored.

I'll need to come up with a different way to test this part.